Updates to Graph and MSAL libraries and Markdown content updates

irvinesunday opened the pull request:
This PR makes the following changes:

- Updates from Android to Java Graph SDK `com.microsoft.graph:microsoft-graph:1.2.0` and MSAL SDK `com.microsoft.identity.client:msal:0.2.2`
- Content updates to README to include app registrations with the new Azure portal app registrations experience.

Commit:
Switching from Android Graph to Java Graph SDK

Commit:
Adding required properties of the Permission request body

Commit:
Ensuring SendMailSuccessUI runs on own thread

Commit:
MSAL lib. update

Commit:
Removing redundant MsalClientException error handler

Commit:
Replacing redundant syntax and redundant 'imports'

Commit:
Adding README images

Commit:
Updates to README

titusgicheru commented on README.md:
```diff
@@ -29,18 +29,26 @@ To use the Connect sample for Android, you need the following:
+
+    ![A screenshot of Application Id](./readme-images/aad-application-id.PNG)
+
+1. Still on the app page, select **Authentication**. Locate the section **Redirect URIs**. In the _Suggested Redirect URIs for public clients(mobile,desktop)_, check the second box so that the app can work with the MSAL libraries used in the application. (The box should contain the option _msal885914a5-96d3-48b7-80dc-48196fb9683c://auth_). Choose **Save**.
```
You might want to replace the actual Client ID with a placeholder  

> _msal885914a5-96d3-48b7-80dc-48196fb9683c://auth_

to  

> _msalEnter_Your_ClientId

irvinesunday replied:
Thanks. This is done

andrueastman commented on app/src/main/java/com/microsoft/graph/connect/GraphServiceClientManager.java:
```diff
@@ -70,9 +70,9 @@ public synchronized IGraphServiceClient getGraphServiceClient(IAuthenticationPro
+            mGraphServiceClient = GraphServiceClient.fromConfig(clientConfig);
         }
-
+// GraphServiceClient.Builder().fromConfig(clientConfig).buildClient();
```
Lets remove this line if we can since it is not being used.

irvinesunday replied:
Done. Thanks.

andrueastman commented on app/src/main/java/com/microsoft/graph/connect/SendMailActivity.java:
```diff
@@ -292,24 +288,34 @@ private void resetUIForSendMail() {
-                SendMailActivity.this,
-                R.string.send_mail_toast_text_error,
-                Toast.LENGTH_LONG).show();
+        runOnUiThread(new Runnable() {
```
👍

Commit:
Removing redundant line of commented code

Commit:
Masking the Client Id

Commit:
Removing a 'left-over' Client Id character

Commit:
Adding class and interface to help read ClientID

Commit:
Adding Msal/MS Graph Android wrapper

Commit:
Implementing the manifest class and interface

Commit:
Getting Client ID

Commit:
Formatting changes

Commit:
Refactored to use the Android wrapper

Commit:
Space formatting

NakulSabharwal commented on app/build.gradle:
```diff
@@ -34,29 +33,28 @@ android {
-    compile 'com.microsoft.graph:msgraph-sdk-android:1.3.2'
-
+    implementation 'com.microsoft.graph:microsoft-graph:1.2.0'
+    implementation 'com.microsoft.identity.client:msal:0.2.2'
```
This is not required since you are using android auth already

irvinesunday replied:
Removed. Thanks

NakulSabharwal commented on app/src/main/java/com/microsoft/graph/connect/GraphServiceClientManager.java:
```diff
@@ -67,12 +77,19 @@ public synchronized IGraphServiceClient getGraphServiceClient() {
+                    getAppActivity(),
+                    GraphServiceClientManager.getApp(),
+                    mAuthenticationManager.getPublicClient(),
+                    Constants.SCOPES);
```
Have msalAuthenticationProvider as a member variable and init only when its null. Same with graphClient.

irvinesunday replied:
Done! Thanks

NakulSabharwal commented on app/src/main/java/com/microsoft/graph/connect/util/ManifestReader.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Override
+    public int getApplicationMetadataValueInt(String key) {
```
Are all these functions actually required ? Can if not use the functions that are actually be used in place where they are being called.

irvinesunday replied:
Removed the unused functions.

NLLAPPS commented:
Hi @NakulSabharwal and @irvinesunday I have been following this repo closely as I am changing my implementation from  OneDrive SDK to MSGraph.

@irvinesunday  you have removed msgraph-sdk-android-aut in pull#20 but just re added ascom.microsoft.graph:microsoft-graph-android-auth:0.1.0-SNAPSHOT

I had followed change at pull#20 and only using 

    implementation 'com.microsoft.identity.client:msal:0.3.+'
    implementation 'com.microsoft.graph:microsoft-graph:1.2.0'

this allows me to use onActivityResult. which microsoft-graph-android-auth only supported with Activity (I needed to use it in fragment and had to pass it from activity to the actual fragment)

How does all this come together? What should I use?

Thanks

Commit:
Removing MSAL Library

Removing reference to MSAL lib. since we are using the Android auth wrapper for authentication

Commit:
Removing unused code

NakulSabharwal commented:
> Hi @NakulSabharwal and @irvinesunday I have been following this repo closely as I am changing my implementation from OneDrive SDK to MSGraph.
> 
> @irvinesunday you have removed msgraph-sdk-android-aut in pull#20 but just re added ascom.microsoft.graph:microsoft-graph-android-auth:0.1.0-SNAPSHOT
> 
> I had followed change at pull#20 and only using
> 
> ```
> implementation 'com.microsoft.identity.client:msal:0.3.+'
> implementation 'com.microsoft.graph:microsoft-graph:1.2.0'
> ```
> 
> this allows me to use onActivityResult. which microsoft-graph-android-auth only supported with Activity (I needed to use it in fragment and had to pass it from activity to the actual fragment)
> 
> How does all this come together? What should I use?
> 
> Thanks

Hey @NLLAPPS ,  you can use both msal or android-auth in your project, while using android-auth you need to pass instance of activity which is creating that graph client, for the remaining activities you dont need to pass any instance.

NLLAPPS commented:
Thanks! Using msal allows me to stop onActivityResult which is great.

What I am trying to understand is what us the reason for android-auto's existence of masal java sdk is working fine on android?

Commit:
Code refactoring
